### PR TITLE
add the dataspace role

### PR DESCRIPTION
**Pull request by kayiwa**
clean up the shibboleth contact user
add dspace tests
add dataspace staging VM

### CODE REVIEW

**Comment by kayiwa** [comment]
this closes #1621

File: group_vars/dataspace/dev.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+nginx_install_from: nginx_repository
+deploy_user: "dspace"
+deploy_user_uid: 1005
+java_type: jdk
+tomcat_user: "{{ deploy_user }}"
+tomcat_group: "{{ deploy_user }}"
+tomcat_install_path: "/home/{{ deploy_user }}"
+dspace_db_user: "oai_db_user"
+dspace_db_password: "{{ vault_dataspace_dev_db_password }}"
+dspace_db_name: "oai_staging_db"
```
Comment (jrgriffiniii) [inline group_vars/dataspace/dev.yml:16]: I approved of this for the OAR with https://github.com/pulibrary/princeton_ansible/pull/1624, but I just to confirm, there isn't going to conflict with the planned `staging` environment, correct?

File: group_vars/dataspace/production.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    value: "{{ postgres_host }}"
+  - name: PGDATABASE
+    value: "{{ dspace_db_name }}"
+  - name: PGUSER
+    value: "{{ dspace_db_user }}"
+  - name: PGPORT
+    value: "{{ dspace_db_port }}"
+  - name: PGPASSWORD
+    value: "{{ dspace_db_password }}"
+shib_hostname: "dataspace.pulcloud.io"
```
Comment (jrgriffiniii) [inline group_vars/dataspace/production.yml:67]: This will need to be updated following the final production release, correct?

File: group_vars/oar/production.yml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    value: "{{ postgres_host }}"
+  - name: PGDATABASE
+    value: "{{ dspace_db_name }}"
+  - name: PGUSER
+    value: "{{ dspace_db_user }}"
+  - name: PGPORT
+    value: "{{ dspace_db_port }}"
+  - name: PGPASSWORD
+    value: "{{ dspace_db_password }}"
+shib_hostname: "oar.pulcloud.io"
```
Comment (jrgriffiniii) [inline group_vars/oar/production.yml:68]: Please see https://github.com/pulibrary/princeton_ansible/pull/1668#pullrequestreview-466046817

File: host_vars/gcp_postgres_prod1.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+---
+pg_enabled: false
+node_identifier: 1
+repmgr_master: true
+standby_db_host: gcp-postgres-prod2
+master_db_host: gcp-postgres-prod1
```
Comment (jrgriffiniii) [inline host_vars/gcp_postgres_prod1.yml:6]: Maybe `leader_db_host` here instead please? I am sorry, this can be a separate PR, but I just wanted to note this.

File: roles/pulibrary.dataspace/handlers/main.yml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+---
+# handlers file for roles/pulibrary.oar
```
Comment (jrgriffiniii) [inline roles/pulibrary.dataspace/handlers/main.yml:2]: Should this comment be `# handlers file for roles/pulibrary.dataspace`?

File: roles/pulibrary.dataspace/vars/main.yml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+---
+# vars file for roles/pulibrary.oar
```
Comment (jrgriffiniii) [inline roles/pulibrary.dataspace/vars/main.yml:2]: This should read `roles/pulibrary.dataspace`

File: roles/pulibrary.oar/templates/shibboleth2.xml.j2 [hunk 7]
```diff
@@ -60,7 +60,7 @@
         also add your own attributes with values that can be plugged into the
         templates, e.g., helpLocation below.
         -->
-        <Errors supportContact="root@localhost"
+        <Errors supportContact="dspadmin@princeton.edu"
```
Comment (jrgriffiniii) [inline roles/pulibrary.oar/templates/shibboleth2.xml.j2:63]: 👍

File: group_vars/dataspace/production.yml (hunk 2 above)
Comment (kayiwa) [reply group_vars/dataspace/production.yml:67]: that will be a separate PR

File: host_vars/gcp_postgres_prod1.yml (hunk 4 above)
Comment (kayiwa) [reply host_vars/gcp_postgres_prod1.yml:6]: I will issue and make this a PR. It happens in more than just one place.

File: group_vars/dataspace/dev.yml (hunk 1 above)
Comment (kayiwa) [reply group_vars/dataspace/dev.yml:16]: it's been a mistake since we implemented this. and a confusing one...

File: roles/pulibrary.dataspace/vars/main.yml (hunk 6 above)
Comment (kayiwa) [reply roles/pulibrary.dataspace/vars/main.yml]: fixed here. https://github.com/pulibrary/princeton_ansible/pull/1668/commits/2701ec473819003eb2f9ad6d8718eda6807b623e

File: roles/pulibrary.dataspace/handlers/main.yml (hunk 5 above)
Comment (kayiwa) [reply roles/pulibrary.dataspace/handlers/main.yml]: fixed here. Thanks https://github.com/pulibrary/princeton_ansible/pull/1668/commits/2701ec473819003eb2f9ad6d8718eda6807b623e

File: host_vars/gcp_postgres_prod1.yml (hunk 4 above)
Comment (kayiwa) [reply host_vars/gcp_postgres_prod1.yml:6]: Created an issue here https://github.com/pulibrary/princeton_ansible/issues/1673
